src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -1130,6 +1160,89 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_to_days() {
+        let cases = vec![
+            ("950501", 728779),
+            ("2007-10-07", 733321),
+            ("2008-10-07", 733687),
+            ("08-10-07", 733687),
+            ("0000-01-01", 1),
+            ("2007-10-07 00:00:59", 733321),
+        ];
+        let mut ctx = EvalContext::default();
+        for (arg, exp) in cases {
+            test_ok_case_one_arg(
+                &mut ctx,
+                ScalarFuncSig::ToDays,
+                Datum::Time(Time::parse_utc_datetime(arg, 6).unwrap()),
+                Datum::I64(exp),
+            );
+        }
+
+        // test NULL case
+        test_err_case_one_arg(&mut ctx, ScalarFuncSig::ToDays, Datum::Null);
+
+        // test ZERO case
+        test_err_case_one_arg(
+            &mut ctx,
+            ScalarFuncSig::ToDays,
+            Datum::Time(Time::parse_utc_datetime("0000-00-00 00:00:00", 6).unwrap()),
+        );
+    }
+
+    #[test]
+    fn test_date_diff() {
+        let cases = vec![
+            (
+                "0000-01-01 00:00:00.000000",
+                "0000-01-01 00:00:00.000000",
+                0,
+            ),
+            (
+                "2018-02-01 00:00:00.000000",
+                "2018-02-01 00:00:00.000000",
+                0,
+            ),
+            (
+                "2018-02-02 00:00:00.000000",
+                "2018-02-01 00:00:00.000000",
+                1,
+            ),
+            (
+                "2018-02-01 00:00:00.000000",
+                "2018-02-02 00:00:00.000000",
+                -1,
+            ),
+            (
+                "2018-02-02 00:00:00.000000",
+                "2018-02-01 23:59:59.999999",
+                1,
+            ),
+            (
+                "2018-02-01 23:59:59.999999",
+                "2018-02-02 00:00:00.000000",
```

Comment:
The following case in mysql:
```
SELECT SUBTIME('2007-12-31 23:59:59.999999','xxx');
+---------------------------------------------+
| SUBTIME('2007-12-31 23:59:59.999999','xxx') |
+---------------------------------------------+
| NULL                                        |
+---------------------------------------------+
1 row in set, 1 warning (0.00 sec)


```
And how about the current PR's result?

Reply:
The result:
```
SELECT SUBTIME('2007-12-31 23:59:59.999999','xxx');
ERROR 1105 (HY000): expected integer
```

It seems that TiKV has not implemented builtinSubStringAndStringSig yet.
I think it has nothing to do with the current PR.

Reply:
Here we implement `sub_duration_and_string` and in this test, we are testing it. What I mean is in `MySQL`,  when we find the second argument is not a legal duration, it will return null. And we need to add this case in the current test. Also it behaviors the same in tidb:
```
// evalDuration evals a builtinSubDurationAndStringSig.
// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_subtime
func (b *builtinSubDurationAndStringSig) evalDuration(row chunk.Row) (types.Duration, bool, error) {
	arg0, isNull, err := b.args[0].EvalDuration(b.ctx, row)
	if isNull || err != nil {
		return types.ZeroDuration, isNull, err
	}
	s, isNull, err := b.args[1].EvalString(b.ctx, row)
	if isNull || err != nil {
		return types.ZeroDuration, isNull, err
	}
	if !isDuration(s) {
		return types.ZeroDuration, true, nil
	}
.....
```